microsoft-azure-storage/src/com/microsoft/azure/storage/blob/BlobOutputStreamInternal.java

```diff
@@ -425,9 +422,9 @@ public Void call() {
 
         // Add future to set
         this.futureSet.add(this.completionService.submit(worker));
-        
+
         // Reset buffer.
-        this.outBuffer = new ByteArrayOutputStream();
+        this.outBuffer = new GettableByteArrayOutputStream();
```

Comment:
>this.outBuffer = new GettableByteArrayOutputStream(); [](start = 8, length = 53)

I think we can actually just call reset() right?

Reply:
We tried that but that actually does not release the acquired bytes in memory, we have to use the sames bytes.